File: linkchat-web/src/types/message.ts
export interface Message {
  id: string
  senderId: string
  senderName: string
  senderAvatar: string
  receiverId: number
  groupId: number | null
  messageType: 'TEXT' | 'IMAGE' | 'FILE' | 'EMOJI' | 'SYSTEM'
  content: string
  fileUrl: string
  fileName: string
  fileSize: number
  fileMime?: string
  isRecalled: number
  quotedMsgId: number | null
  ackStatus: 'SENT' | 'DELIVERED' | 'READ' | 'FAILED'
  createdAt: string
}

export interface SendMessageRequest {
  receiverId: number
  groupId?: number
  messageType: string
  content?: string
  fileUrl?: string
  fileName?: string
  fileSize?: number
  fileMime?: string
  quotedMsgId?: number
}

File: linkchat-server/src/main/java/com/linkchat/server/entity/Message.java
package com.linkchat.server.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("message")
public class Message {
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;
    private Long senderId;
    private Long receiverId;
    private Long groupId;
    private String messageType;
    private String content;
    private String fileUrl;
    private String fileName;
    private Long fileSize;
    private String fileMime;
    private Integer isRecalled;
    private Integer isDeleted;
    private Long quotedMsgId;
    private String ackStatus;
    private Integer retryCount;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}